File: src/RunConfigImporterDefaultComponent.java
import com.intellij.openapi.components.ProjectComponent;
import com.intellij.openapi.project.Project;
import impl.RunConfImporter;

import java.io.File;

public class RunConfigImporterDefaultComponent implements ProjectComponent {

	private Project project;

	RunConfigImporterDefaultComponent(Project project) {
		this.project = project;
	}

	@Override
	public void projectOpened() {
		ImportConfig importConfig = ImportConfig.getInstance(project);
		assert importConfig != null;
		if (importConfig.hasDefault()) {
			String path = importConfig.getDefaultPath();
			File file = new File(path);
			if (file.exists()) {
				RunConfImporter.getInstance(project).importConfigs(path);
			}
		}
	}
}

File: src/RunConfigImporterDefaultComponent.kt
import com.intellij.openapi.components.ProjectComponent
import com.intellij.openapi.project.Project
import impl.RunConfImporter

import java.io.File

class RunConfigImporterDefaultComponent internal constructor(private val project: Project) : ProjectComponent {

    override fun projectOpened() {
        val appConfig = ImportConfigApp.getInstance()
        importFromPath(appConfig)
        val importConfig = ImportConfig.getInstance(project)
        importFromPath(importConfig)
    }

    private fun importFromPath(importconfig: ImportConfigInterface) {
        if (importconfig.hasDefault() && importconfig.getPath() != null) {
            val path = importconfig.getPath()
            val dir = File(path!!)
            if (dir.exists()) {
                RunConfImporter.getInstance(project).importConfigs(path)
            }
        }
    }
}
